test/ruby/test_object.rb

```diff
@@ -422,6 +422,18 @@ def test_check_to_integer
     assert_equal(1+3+5+7+9, n)
   end
 
+  def test_max_shape_variation_with_performance_warnings
+    assert_in_out_err([], <<-INPUT, %w(), /Maximum shapes variations \(8\) reached by Foo, instance variables accesses will be slower\.$/)
+      $VERBOSE = true
```

Comment:
Does it also work with `$VERBOSE = false`? See https://github.com/ruby/ruby/pull/7715#discussion_r1169777099
If only with `$VERBOSE = true`, it seems too complicated to enable performance warnings (need to enable two things, very easy to mess up)

Reply:
I believe it needs both yes.